Add job to run testsuite with MSSQL on Windows

ADmad opened the pull request:
Does anybody have ideas how to fix the failing tests for CURL and currency formatting (related to ICU) on Windows?
<!---

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

The best way to propose a feature is to open an issue first and discuss your ideas there before implementing them.

Always follow the [contribution guidelines](https://github.com/cakephp/cakephp/blob/master/.github/CONTRIBUTING.md) when submitting a pull request. In particular, make sure existing tests still pass, and add tests for all new behavior. When fixing a bug, you may want to add a test to verify the fix.

-->

ADmad committed:
Add job to run testsuite with MSSQL on Windows

garas commented:
CLDR 36 has new `en_IN` currency format in [CLDR-13097](https://unicode-org.atlassian.net/browse/CLDR-13097). Tests on Linux will start to fail too once ICU/CLDR will be updated.

Curl tests expects not empty content on `http://localhost`, but server responds with empty content. It is unclear what server listens on port 80 on Github Actions Windows VM. Need some more reliable target server, or to skip test.

ADmad committed:
Update tests relying en_IN currency pattern as it differs between ICU/CLDR versions.

ADmad committed:
Update Curl adapter tests to not fail on GH workflow's Windows VM.

ADmad commented:
> CLDR 36 has new en_IN currency format in CLDR-13097. Tests on Linux will start to fail too once ICU/CLDR will be updated.

I updated the tests to not use `en_IN` currency format.

> Curl tests expects not empty content on http://localhost, but server responds with empty content. It is unclear what server listens on port 80 on Github Actions Windows VM. Need some more reliable target server, or to skip test.

I removed the assertion for non empty content. Lack of exception means the request was successful and there's also an assertion for non empty headers which does pass.

Also updated the test for `http://dummy/?sleep` URL to account for both possible error messages on linux and windows.

ADmad commented:
> Tests: 10129, Assertions: 28788, Skipped: 211, Incomplete: 5

On AppVeyor

> Tests: 10129, Assertions: 29998, Skipped: 179, Incomplete: 5.

On Github, so 32 more tests run :slightly_smiling_face:

othercorey commented on .github/workflows/ci.yml:
```diff
@@ -78,6 +78,38 @@ jobs:
 
+  testsuite-windows:
+    runs-on: windows-2019
+    name: Windows - PHP 7.4 & mssql
```
```suggestion
    name: Windows - PHP 7.4 & SQL Server
```

othercorey commented:
I can't seem to find what version of SQL Server is provided in the `windows-2019` image. I see the various add-on packages, but not the actual server version.

ADmad commented:
@othercorey https://github.com/actions/virtual-environments/issues/413#issuecomment-589949674

ADmad committed:
Cleanup CI workflow config

ADmad committed:
Show the SQL Server info
